packages/block-library/src/gallery/gallery.native.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+import { isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import GalleryImage from './gallery-image';
+import { defaultColumnsNumber } from './shared';
+import styles from './gallery-styles.scss';
+import Tiles from './tiles';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Caption } from '@wordpress/block-editor';
+import { useState } from '@wordpress/element';
+
+const TILE_SPACING = 15;
+
+// we must limit displayed columns since readable content max-width is 580px
+const MAX_DISPLAYED_COLUMNS = 4;
+const MAX_DISPLAYED_COLUMNS_MOBILE = 2;
+
+export const Gallery = ( props ) => {
+	const [ isCaptionSelected, setIsCaptionSelected ] = useState( false );
+
+	const {
+		clientId,
+		selectedImage,
+		mediaPlaceholder,
+		onBlur,
+		onMoveBackward,
+		onMoveForward,
+		onRemoveImage,
+		onSelectImage,
+		onSetImageAttributes,
+		onFocusGalleryCaption,
+		attributes,
+		isSelected,
+		isMobile,
+		onFocus,
+	} = props;
+
+	const {
+		columns = defaultColumnsNumber( attributes ),
+		imageCrop,
+		images,
+	} = attributes;
+
+	// limit displayed columns based on viewport width
+	const displayedColumns = isMobile ?
```

Comment:
Should we change `isMobile` as `isMobilePortrait` ? Because this is false for mobile landscape.

Reply:
Good point! `isMobile` is not the best name for this. I guess even using the name mobile in the the name at all can be misleading, since currently it's conditioned on viewport width being less than 600, while some larger tablets have a portrait viewport wider than that.

In an earlier iteration (when there were _3_ possibilities for displayed columns), I had passed another flag called `isNarrow`, which I think works semantically, but at that time may have been ambiguous (it wasn't clearly _more_ or _less_ narrow than `isMobile` for example). Since we're not using that anymore (and limiting to 4 columns on mobile in general because of the limited readable content width), maybe we could reuse that moniker? I.e. we could use the name `isNarrow: "< small"` (viewport width < 600), in place of `isMobile`? Wdyt?

**tl;dr:** Do you think `isNarrow` could work here?

Reply:
`isNarrow` sounds OK to me.

Reply:
It would really help to put a code comment somewhere about what range corresponds to isNarrow.

Reply:
Good suggestion! I'll add a comment too. 😄